prql-compiler/src/sql/gen_expr.rs

```diff
@@ -184,6 +183,60 @@ pub(super) fn translate_literal(l: Literal, ctx: &Context) -> Result<sql_ast::Ex
     })
 }
 
+fn translate_datetime_literal(
+    data_type: sql_ast::DataType,
+    value: String,
+    ctx: &Context,
+) -> sql_ast::Expr {
+    if ctx.dialect.is::<crate::sql::dialect::SQLiteDialect>() {
+        translate_datetime_literal_with_sqlite_function(data_type, value)
+    } else {
+        translate_dateatime_literal_with_typed_string(data_type, value)
+    }
+}
+
+fn translate_dateatime_literal_with_typed_string(
+    data_type: sql_ast::DataType,
+    value: String,
+) -> sql_ast::Expr {
+    sql_ast::Expr::TypedString { data_type, value }
+}
+
+fn translate_datetime_literal_with_sqlite_function(
+    data_type: sql_ast::DataType,
+    value: String,
+) -> sql_ast::Expr {
+    let timezone_indicator_regex = Regex::new(r"([+-]\d{2}):?(\d{2})$").unwrap();
```

Comment:
```suggestion
    // TODO: promote parsing timezone handling to the parser; we should be storing
    // structured data rather than strings in the AST
    let timezone_indicator_regex = Regex::new(r"([+-]\d{2}):?(\d{2})$").unwrap();
```

(no need to do as part of this tbc)

Reply:
yeah I agree we should try parsing the timezone information from user input to provide more flexible transpiling.